phone/src/os/nui-events/hooks/useNuiFetch.ts

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useRef, useState } from 'react';
+
+interface IAbortableFetch {
+  abort: () => void;
+  ready: Promise<Response>;
+}
+
+function abortableFetch(request, opts): IAbortableFetch {
+  const controller = new AbortController();
+  const signal = controller.signal;
+
+  return {
+    abort: () => controller.abort(),
+    ready: fetch(request, { ...opts, signal }),
+  };
+}
+
+type FetchData = Record<string, any> | boolean | null | undefined | string | number;
+
+type UseNuiFetchResponse<I, R> = [
+  (data: I) => void,
+  { response: R; error: Error | unknown; loading: boolean },
+];
+
+const DEFAULT_HANDLER = () => {};
+
+export function useNuiFetch<I = FetchData, R = FetchData>(
+  event: string,
+  handler: (d?: R) => void = DEFAULT_HANDLER,
+): UseNuiFetchResponse<I, R> {
```

Comment:
Can you add some comments on what's going on here? Would be nice to have some jsdocs :p